solana-programs/anchor/audius-data/tests/audius-data.ts

```diff
@@ -882,18 +710,95 @@ describe("audius-data", function () {
+        userStorageAccount: delegate.userAccountPDA,
+        userAuthorityKeypair: delegate.userAuthorityDelegateKeypair,
+        userAuthorityDelegate: delegate.userAuthorityDelegatePDA,
+        authorityDelegationStatusAccount: delegate.authorityDelegationStatusPDA,
       })
     )
       .to.eventually.be.rejected.and.property("msg")
       .to.include(`This authority's delegation status is revoked.`);
   });
 
+  it("delegate adds/removes another delegate", async function () {
+    const firstDelegate = await testCreateUserDelegate({
+      adminKeypair,
+      adminStorageKeypair,
+      program,
+      provider,
+      ...getURSMParams(),
+    });
+
+    const secondDelegate = await testCreateUserDelegate({
+      adminKeypair,
+      adminStorageKeypair,
+      program,
+      provider,
+      ...getURSMParams(),
+    });
+
+    const userAuthorityDelegateSeeds = [
+      firstDelegate.userAccountPDA.toBytes().slice(0, 32),
+      secondDelegate.userAuthorityDelegateKeypair.publicKey
+        .toBytes()
+        .slice(0, 32),
+    ];
+    const res = await PublicKey.findProgramAddress(
+      userAuthorityDelegateSeeds,
+      program.programId
+    );
+    const newUserAuthorityDelegatePDA = res[0];
+    const newUserAuthorityDelegateBump = res[1];
+
+    const addUserAuthorityDelegateArgs = {
+      accounts: {
+        admin: adminStorageKeypair.publicKey,
+        user: firstDelegate.userAccountPDA,
+        newUserAuthorityDelegate: newUserAuthorityDelegatePDA,
+        signerUserAuthorityDelegate: firstDelegate.userAuthorityDelegatePDA,
+        authorityDelegationStatus: firstDelegate.authorityDelegationStatusPDA,
+        authority: firstDelegate.userAuthorityDelegateKeypair.publicKey,
+        payer: provider.wallet.publicKey,
+        systemProgram: SystemProgram.programId,
+      },
+      signers: [firstDelegate.userAuthorityDelegateKeypair], // the first user's delegate signs
+    };
+
+    await program.rpc.addUserAuthorityDelegate(
+      firstDelegate.baseAuthorityAccount,
+      firstDelegate.handleBytesArray,
+      firstDelegate.userBumpSeed,
+      secondDelegate.userAuthorityDelegateKeypair.publicKey, // seed for userAuthorityDelegatePda
+      addUserAuthorityDelegateArgs
+    );
+
+    const removeUserAuthorityDelegateArgs = {
+      accounts: {
+        admin: adminStorageKeypair.publicKey,
+        user: firstDelegate.userAccountPDA,
+        newUserAuthorityDelegate: newUserAuthorityDelegatePDA,
+        signerUserAuthorityDelegate: firstDelegate.userAuthorityDelegatePDA,
+        authorityDelegationStatus: firstDelegate.authorityDelegationStatusPDA,
+        authority: firstDelegate.userAuthorityDelegateKeypair.publicKey,
+        payer: provider.wallet.publicKey,
+        systemProgram: SystemProgram.programId,
+      },
+      signers: [firstDelegate.userAuthorityDelegateKeypair],
+    };
+
+    await program.rpc.removeUserAuthorityDelegate(
```

Comment:
1) should we also assert, for this test, that second can('t) remove first? 
2) for this it doesn't look like we're doing any assertions - we should assert that added delegate can do actions, or that the delegate authority status PDA is now equal to isRevoked false right? It's not clear from the setup of the test that this outcome actually changed just bc the RPC method didn't throw an error.